diskqueue.go

```diff
@@ -402,17 +454,25 @@ func (d *diskQueue) writeOne(data []byte) error {
 		return err
 	}
 
-	totalBytes := int64(4 + dataLen)
 	d.writePos += totalBytes
```

Comment:
Do we need to add numFileMsgsBytes to d.writePos explicitly if numFileMsgsBytes is written into the writeBuf?

Reply:
Nope, d.maxBytesPerFileRead can be set as d.writePos if [readFileNum == writeFileNum and we write a complete file](https://github.com/kev1n80/go-diskqueue/blob/DepthImpl/diskqueue.go#L468). When DiskQueue normally reads through data files, we do not want DiskQueue to read the num of msgs in this file. We only want to grab this data when we are reading an incomplete file.

Reply:
Got it.

Btw, later maybe we need to consider how to read [the last 8 bytes](https://github.com/kev1n80/go-diskqueue/blob/DepthImpl/diskqueue.go#L443) out while passing checkTailCorruption function.

Because, in https://github.com/kev1n80/go-diskqueue/blob/DepthImpl/diskqueue.go#L593
```
if d.readFileNum < d.writeFileNum || d.readPos < d.writePos {
		return
} 
```
Since d.writePos does not contain the last 8 bytes, the readPos could not read the last 8 bytes if `d.readFileNum == d.writeFileNum`.

Reply:
Thank you for the consideration! As per our conversation, this might not be an issue since `readPos` is not used to read the last 8 bytes of a file. However, this might cause new corner cases to arise.